bigquery_etl/query_scheduling/task.py

```diff
@@ -467,12 +460,27 @@ def of_python_script(cls, query_file, metadata=None, dag_collection=None):
         return task
 
     @classmethod
-    def of_dq_check(cls, query_file, metadata=None, dag_collection=None):
+    def of_dq_check(cls, query_file, is_check_fail, metadata=None, dag_collection=None):
         """Create a task that schedules DQ check file in Airflow."""
         task = cls.of_query(query_file, metadata, dag_collection)
         task.query_file_path = query_file
         task.is_dq_check = True
+        task.is_dq_check_fail = is_check_fail
         task.depends_on_fivetran = []
+        if task.is_dq_check_fail:
+            task.task_name = (
+                f"checks__fail_{task.dataset}__{task.table}__{task.version}"[
+                    -MAX_TASK_NAME_LENGTH:
```

Comment:
If I recall correctly we shouldn't have to truncate the task anymore since Airflow should do this automatically, but perhaps to be on the safer side we should still keep it.